File: modules/dnd-status/src/DndStatusModule.ts
import { NativeModule, requireNativeModule } from "expo";

declare class DndStatusModuleType extends NativeModule<{}> {
  isDndActive(): boolean;
  isSupported(): boolean;
  isNotificationPolicyAccessGranted(): boolean;
  setDndActive(active: boolean): boolean;
  openDndSettings(): void;
}

let DndStatusModule: DndStatusModuleType;

try {
  DndStatusModule = requireNativeModule<DndStatusModuleType>("DndStatus");
} catch {
  DndStatusModule = {
    isDndActive: () => false,
    isSupported: () => false,
    isNotificationPolicyAccessGranted: () => false,
    setDndActive: () => false,
    openDndSettings: () => {},
    addListener: () => ({ remove: () => {} }),
    removeListeners: () => {},
  } as unknown as DndStatusModuleType;
}

export default DndStatusModule;

File: apps/mobile/modules/dnd-status/ios/DndStatusModule.swift
import ExpoModulesCore

public class DndStatusModule: Module {
  public func definition() -> ModuleDefinition {
    Name("DndStatus")

    Function("isDndActive") { () -> Bool in
      return false
    }

    Function("isSupported") { () -> Bool in
      return false
    }

    Function("isNotificationPolicyAccessGranted") { () -> Bool in
      return false
    }

    Function("setDndActive") { (active: Bool) -> Bool in
      return false
    }

    Function("openDndSettings") { () -> Void in
      // No-op on iOS
    }
  }
}
